File: longest_word/longest_word.rb
def longest_word(sentence)
  # replace non-alpha characters with spaces, reduce all multi-spaces to just
  # one space, strip any leading or trailing spaces, and then split words
  # into an array
  sentence = sentence.gsub(/[^a-zA-Z]/,' ').squeeze(' ').strip.split(' ')
  longest_word_size = 0;
  longest_word = ""
  # loop over each word and determine if the word size is longer than the
  # longest word size
  sentence.each do |word|
    if word.size > longest_word_size
      longest_word = word
      longest_word_size = word.size
    end
  end
  return longest_word
end

puts longest_word("hellooooo!!**#world")

File: longest_word/longest_word.js
function longest_word(sentence) {
  // since the "word" does not include non-alpha characters, replace all
  // non-alpha characters with spaces
  var fixed_sentence = sentence.replace(/[^a-zA-Z]/g,' ');
  var longest_word = "";
  var longest_word_size = 0;
  var count = 0;
  var start = 0;
  var length = fixed_sentence.length;
  // loop over sentence and if there is a space, or if you reach the end of 
  // the word, then check if the current word is longer than the longest_word
  // if it is, create a substring that is the longest word
  for (var i = 0; i < length; i++) {
    if (fixed_sentence[i] == ' ' || (i + 1) == length) {
      if (count > longest_word_size) {
        longest_word_size = count;
        longest_word = fixed_sentence.substring(start, i);
      }
      start = i + 1;
      count = 0;
    }
    count++;
  }
  return longest_word;
}

console.log(longest_word("hellooooo!!**#world"))